pkg/opensearch/stream_handler.go

```diff
@@ -0,0 +1,168 @@
+	return &backend.SubscribeStreamResponse{
+		Status: backend.SubscribeStreamStatusOK,
+	}, nil
+}
+
+func (o *OpenSearchDatasource) RunStream(ctx context.Context, req *backend.RunStreamRequest, sender *backend.StreamSender) error {
+	o.logger.Info("RunStream (Polling Mode) called", "path", req.Path)
+	o.logger.Debug("RunStream", "full_req_details", req)
+
+	if !strings.HasPrefix(req.Path, "tail/") {
+		o.logger.Error("RunStream: invalid path format", "path", req.Path)
+		return fmt.Errorf("RunStream: invalid path format for stream: %s, expected 'tail/{refId}'", req.Path)
+	}
+	refId := strings.TrimPrefix(req.Path, "tail/")
+	if refId == "" {
+		o.logger.Error("RunStream: missing refId in path after trim", "path", req.Path)
+		return fmt.Errorf("RunStream: missing refId in stream path after trim: %s", req.Path)
+	}
+
+	val, ok := o.streamQueries.Load(refId)
+	if !ok {
+		o.logger.Error("RunStream: query not found in streamQueries map", "refId", refId, "path", req.Path)
+		return fmt.Errorf("query with refId %s not found for streaming. Was it registered?", refId)
+	}
+
+	defer func() {
+		o.logger.Info("RunStream: deleting query from map", "refId", refId)
+		o.streamQueries.Delete(refId)
+	}()
+
+	rawQueryJSON, ok := val.(json.RawMessage)
+	if !ok {
+		o.logger.Error("RunStream: failed to assert query type from map", "refId", refId, "type", fmt.Sprintf("%T", val))
+		return fmt.Errorf("failed to assert query type for refId: %s", refId)
+	}
+
+	o.logger.Info("RunStream: starting polling for query", "refId", refId, "rawQuery", string(rawQueryJSON))
+
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	lastTo := time.Now()
+
+	for {
+		select {
+		case <-ctx.Done():
+			o.logger.Info("RunStream: context canceled, stopping stream", "refId", refId)
+			return ctx.Err()
+		case <-ticker.C:
+			currentTime := time.Now()
+			backendQuery := backend.DataQuery{
+				RefID:     refId,
+				TimeRange: backend.TimeRange{From: lastTo, To: currentTime},
+				JSON:      rawQueryJSON,
+			}
+
+			if !backendQuery.TimeRange.To.After(backendQuery.TimeRange.From) {
+				o.logger.Debug("RunStream: 'To' time is not after 'From' time, skipping poll to avoid empty range", "from", backendQuery.TimeRange.From, "to", backendQuery.TimeRange.To)
+				continue
+			}
+
+			o.logger.Info("RunStream: Polling OpenSearch", "refId", refId, "from", backendQuery.TimeRange.From, "to", backendQuery.TimeRange.To)
+
+			osClient, err := client.NewClient(ctx, req.PluginContext.DataSourceInstanceSettings, o.httpClient, &backendQuery.TimeRange)
+			if err != nil {
+				o.logger.Error("RunStream: failed to create OpenSearch client for poll", "refId", refId, "error", err)
+				continue
+			}
+
+			queryExecutor := newQueryRequest(osClient, []backend.DataQuery{backendQuery}, req.PluginContext.DataSourceInstanceSettings)
+			queryDataResponse, err := queryExecutor.execute(ctx)
+
+			if err != nil {
```

Comment:
Query response returned contains an .Error field instead of returning an error, so this error wouldn't get logged or passed down to the user